internal/strings/strings_test.go

```diff
@@ -89,3 +92,50 @@ func TestRandomStringConcurrency(t *testing.T) {
 		go RandomString(10000)
 	}
 }
+
+func TestAsciiToLower(t *testing.T) {
+	word := "HELLO WORLD"
+	if want, have := "hello world", AsciiToLower(word); want != have {
+		t.Errorf("Unexpected lowercase word, want %q, have %q", want, have)
+	}
+}
+
+func BenchmarkAsciiVsUnicodeCaseString(b *testing.B) {
+	strs := []string{
+		"!!!!!!!!!!!!!!aaaaaaaaa please merge me!!!",
+		"This is a String With a lot of Characters!!!",
+		"this is a lowercase string with many characters",
+		"THIS IS AN UPPERCASE STRING WITH MANY CHARACTERS",
+		"ThIs Is A StRiNg WiTh MiXeD CaSe",
+	}
+	b.Run("hacky ascii", func(b *testing.B) {
+		for _, str := range strs {
+			b.Run(str, func(b *testing.B) {
+				for i := 0; i < b.N; i++ {
+					AsciiToLower(str)
+				}
+			})
+		}
+	})
+	b.Run("standard ascii", func(b *testing.B) {
```

Comment:
IIUC, this case is mostly isolating `WrapUnsafe`, and also missing the optimization for already lowercased strings. We probably don't need to keep this benchmark since these are standard practices in the repo